tests/drivers/regulator/api/src/main.c

```diff
@@ -420,6 +420,81 @@ ZTEST(regulator_api, test_get_voltage_error)
 	zassert_equal(regulator_fake_get_voltage_fake.arg1_val, NULL);
 }
 
+ZTEST(regulator_api, test_count_current_limits_not_implemented)
+{
+	unsigned int ret = regulator_count_current_limits(dummy_reg);
+
+	zassert_equal(ret, 0);
+}
+
+ZTEST(regulator_api, test_count_current_limits_ok)
+{
+	RESET_FAKE(regulator_fake_count_current_limits);
+
+	regulator_fake_count_current_limits_fake.return_val = 10;
+
+	zassert_equal(regulator_count_current_limits(reg0), 10U);
+	zassert_equal(regulator_fake_count_current_limits_fake.arg0_val, reg0);
+	zassert_equal(regulator_fake_count_current_limits_fake.call_count, 1U);
+}
+
+ZTEST(regulator_api, test_count_current_limits_fail)
+{
+	RESET_FAKE(regulator_fake_count_current_limits);
+
+	regulator_fake_count_current_limits_fake.return_val = -EINVAL;
+
+	zassert_equal(regulator_count_current_limits(reg0), -EINVAL);
+	zassert_equal(regulator_fake_count_current_limits_fake.arg0_val, reg0);
+	zassert_equal(regulator_fake_count_current_limits_fake.call_count, 1U);
+}
+
+ZTEST(regulator_api, test_list_current_limit_not_implemented)
```

Comment:
```suggestion
ZTEST(regulator_api, test_list_current_limit_not_supported)
```

Reply:
The other tests use *_not_implemented, wouldn't we prefer to be consistent?

Reply:
fair enough, as it returns 0 because of that.